### ISSUE TITLE
[A11Y] File lists are currently inaccessible ASCII art representations

### ISSUE THREAD

**Issue by crutchcorn** [withastro/docs#2167]
Currently, when we're representing a file tree of some kind, we display something like the following:

![An ASCII representation of a file tree, based on the output of the `tree` command](https://user-images.githubusercontent.com/9100169/207042345-0689b22b-b24c-438b-b205-3aebaca161d0.png)

This doesn't read out well with accessible technologies such as screen-readers. Instead, we might want to consider building out an interactive component that supports assistive technologies better, as well as collapseable states.

An example of what a component might look like could be found in [Ant Design](https://ant.design/components/tree#components-tree-demo-directory) or [Fluent UI](https://developer.microsoft.com/en-us/fluentui#/controls/web/groupedlist)

# List of Impacted Files

- [x] [/old-translations/de/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/old-translations/de/core-concepts/project-structure.md)
- [x] [/old-translations/de/guides/styling.md](https://github.com/withastro/docs/tree/main/old-translations/de/guides/styling.md)
- [x] [/old-translations/es/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/old-translations/es/core-concepts/project-structure.md)
- [ ] [/src/pages/de/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/de/core-concepts/project-structure.md)
- [ ] [/src/pages/de/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/de/core-concepts/routing.md)
- [ ] [/src/pages/de/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/de/install/manual.md)
- [x] [/src/pages/en/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/en/core-concepts/project-structure.md)
- [x] [/src/pages/en/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/en/core-concepts/routing.md)
- [x] [/src/pages/en/guides/cms/contentful.md](https://github.com/withastro/docs/tree/main/src/pages/en/guides/cms/contentful.md)
- [x] [/src/pages/en/guides/cms/storyblok.md](https://github.com/withastro/docs/tree/main/src/pages/en/guides/cms/storyblok.md)
- [x] [/src/pages/en/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/en/install/manual.md)
- [x] [/src/pages/en/reference/publish-to-npm.md](https://github.com/withastro/docs/tree/main/src/pages/en/reference/publish-to-npm.md)
- [ ] [/src/pages/es/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/es/core-concepts/project-structure.md)
- [ ] [/src/pages/es/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/es/core-concepts/routing.md)
- [ ] [/src/pages/es/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/es/install/manual.md)
- [ ] [/src/pages/es/reference/publish-to-npm.md](https://github.com/withastro/docs/tree/main/src/pages/es/reference/publish-to-npm.md)
- [ ] [/src/pages/fr/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/fr/core-concepts/project-structure.md)
- [ ] [/src/pages/fr/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/fr/core-concepts/routing.md)
- [ ] [/src/pages/fr/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/fr/install/manual.md)
- [ ] [/src/pages/fr/reference/publish-to-npm.md](https://github.com/withastro/docs/tree/main/src/pages/fr/reference/publish-to-npm.md)
- [ ] [/src/pages/ja/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/ja/core-concepts/project-structure.md)
- [ ] [/src/pages/ja/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/ja/core-concepts/routing.md)
- [ ] [/src/pages/ja/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/ja/install/manual.md)
- [ ] [/src/pages/pt-br/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/pt-br/core-concepts/project-structure.md)
- [ ] [/src/pages/pt-br/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/pt-br/core-concepts/routing.md)
- [ ] [/src/pages/pt-br/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/pt-br/install/manual.md)
- [ ] [/src/pages/pt-br/reference/publish-to-npm.md](https://github.com/withastro/docs/tree/main/src/pages/pt-br/reference/publish-to-npm.md)
- [ ] [/src/pages/ru/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/ru/core-concepts/project-structure.md)
- [ ] [/src/pages/zh-cn/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/zh-cn/core-concepts/project-structure.md)
- [ ] [/src/pages/zh-cn/core-concepts/routing.md](https://github.com/withastro/docs/tree/main/src/pages/zh-cn/core-concepts/routing.md)
- [ ] [/src/pages/zh-cn/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/zh-cn/install/manual.md)
- [ ] [/src/pages/zh-cn/reference/publish-to-npm.md](https://github.com/withastro/docs/tree/main/src/pages/zh-cn/reference/publish-to-npm.md)
- [ ] [/src/pages/zh-tw/core-concepts/project-structure.md](https://github.com/withastro/docs/tree/main/src/pages/zh-tw/core-concepts/project-structure.md)
- [ ] [/src/pages/zh-tw/install/manual.md](https://github.com/withastro/docs/tree/main/src/pages/zh-tw/install/manual.md)
